force-app/main/default/lwc/nksVideoPlayer/nksVideoPlayer.js

```diff
@@ -0,0 +1,163 @@
+import { LightningElement, api, wire } from 'lwc';
+import getFileType from '@salesforce/apex/NKS_VideoPlayerCtrl.checkFileType';
+import { isVideoFile } from 'c/utils';
+import getVideoTracksInternally from '@salesforce/apex/NKS_VideoPlayerCtrl.getVideoTracksInternally';
+import getVideoTitle from '@salesforce/apex/NKS_VideoPlayerCtrl.getVideoTitle';
+import getStoredThumbnailLink from '@salesforce/apex/NKS_VideoPlayerCtrl.getStoredThumbnailLink';
+import isSandbox from '@salesforce/apex/NKS_VideoPlayerCtrl.isSandbox';
+import addViewCount from '@salesforce/apex/NKS_VideoPlayerCtrl.addViewCount';
+
+export default class NksVideoPlayer extends LightningElement {
+    @api recordId; // Automatically set when within Salesforce
+    videoSrc;
+    videoTitle;
+    thumbnail;
+    context;
+    isFileTypeMp4;
+    tracksAdded = false; // Check whether tracks have been added or not
+    subTracks = [];
+    error;
+
+    connectedCallback() {
+        this.generateVideoPlayer();
+    }
+
+    renderedCallback() {
+        this.attachEventListener();
+
+        if (this.videoSrc && this.subTracks.length > 0 && !this.tracksAdded) {
+            this.addTracksToVideo();
+        }
+    }
+
+    get isExperience() {
+        return this.context === 'Experience';
+    }
+
+    @wire(getFileType, { recordId: '$recordId'})
+    wiredGetFileType(result) {
+        if (result.error) {
+            console.error(result.error);
+        } else if (result.data) {
+            this.isFileTypeMp4 = isVideoFile(result.data);
+            this.filetype = result.data;
+        }
+    }
+
+    async generateVideoPlayer() {
+        try {
+            await this.getVideoIdAndSetVideoUrl();
+            this.subTracks = await this.fetchSubTracks();
+            this.videoTitle = await this.getVideoTitle();
+            this.thumbnail = await this.getThumbnailLink();
+            this.setThumbnail();
+        } catch (error) {
+            this.error = error;
+            console.error(error);
+        }
+    }
+
+    async getVideoIdAndSetVideoUrl() {
+        if (this.recordId) {
+            // Displaying inside Salesforce
+            this.videoSrc = window.location.origin + '/sfc/servlet.shepherd/document/download/' + this.recordId;
+            this.context = 'Standard';
+        } else {
+            // In community
+            const url = window.location.href;
+            const videoId = decodeURIComponent(url.substring(url.lastIndexOf('/') + 1));
+            this.recordId = videoId;
```

Comment:
Litt redd for injection her, har du dobbeltsjekket at denne er håndtert rett i alle queries og andre operasjoner?

Reply:
Bruker binding variable